### ISSUE TITLE
`enw_retrospective_data` potentially not working as expected

### ISSUE THREAD

**Issue by sbfnk** [epiforecasts/epinowcast#79]
Second guessing as there is no function documentation but I'd expect

```r
retro_nat_germany <- enw_retrospective_data(
  nat_germany_hosp,
  rep_days = 30, ref_days = 40
)
```

to return a data set with 30 reporting days and 40 occurrence dates. Instead I get

```r
> length(unique(retro_nat_germany$report_date))
[1] 41
> length(unique(retro_nat_germany$reference_date))
[1] 41
```

It seems to me these two lines should both have a `>` sign, but perhaps I might be misunderstanding what is being done here
https://github.com/epiforecasts/epinowcast/blob/a5616b15857d790eced0f1289b498d42fa0ca6d7/R/preprocess.R#L171
https://github.com/epiforecasts/epinowcast/blob/a5616b15857d790eced0f1289b498d42fa0ca6d7/R/preprocess.R#L176

**Comment by seabbs** [comment]
Yes, the second interpretation is correct with the idea being that you want the state of the data on a given day (driven by report date) and then a certain window of reference days to use to train the model. There is an argument that having both `ref_date` and `ref_days` is a bit much as one is just a helper to get the other. However, personally I think worth the additional complexity due to flexibility but definitely a judgement call.

So I guess this kind of a documentation issue (i.e if current behaviour was documented it would be working as expected). That being said I think setting 40 and getting 40 days of data is probably more in line with user expectations so agree should be updated in both places.
